ring/lifecycler.go

```diff
@@ -585,22 +585,53 @@ func (i *Lifecycler) initRing(ctx context.Context) error {
 			return ringDesc, true, nil
 		}
 
-		// If the ingester failed to clean its ring entry up in can leave its state in LEAVING
+		level.Info(i.logger).Log("msg", "existing entry found in ring", "state", instanceDesc.State, "tokens", len(instanceDesc.Tokens), "ring", i.RingName)
+
+		// If the ingester failed to clean its ring entry up it can leave its state in LEAVING
 		// OR unregister_on_shutdown=false
 		// Move it into ACTIVE to ensure the ingester joins the ring.
```

Comment:
Comment does not match what we do anymore. There's an exception to the rule because of line 600. I wonder why we wouldn't just generate the necessary number of tokens and set state to active - which is what we do if there's no token file.

Reply:
@krajorama I implemented your suggestion, without remembering your suggestion :) I guess it should be a good idea if we both landed on it.